### PR TITLE
ADNI brain classification with Vision Transformer

### PR DESCRIPTION
**Pull request by o-20**
Vision transformer with PyTorch Classifying the ADNI brain dataset, consisting of 1076 samples of 20 slices of 240x256 greyscale image. Upgrades to basic ViT model include a convolutional pre-layer (one of two designs). Data augmentation is used with horizontal, vertical and diagonal flipping to result in 4x as much data.

Consists of four files dataset.py, modules.py, train.py, predict.py. train.py handles the training and predict.py can be used to load a model or data.

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:

- Need more explanation on how VIT architecture works and how you developed your algorithm.
- No plots of training and val accuracy
- Intro seems not finished
- No hyperparameter tuning results are presented in the README.
- No train, val and test split percentage presented in the README while I can see train, val , test split in the code.

Commit messages: clear

Code:
- Uses Vision transformer
- Commented
- Good model design

Functionality/Performance:
- Test accuracy is 68.0%. ("Claimed val ACC is the same as test ACC but no plots or evidence showing that")
- Loss and accuracy plots are unclear (unclear legends).
- Only have validation ACC plot, no training ACC plot 
- No evidence of patient-level split
- Preprocssing data correctly
- No Reference section showing paper, figure and code source.

**Comment by o-20** [comment]
Hi,

If I disagree with these points (eg. no hyperparameter tuning or train, val
test split), how is the best way to communicate it, including it on the
readme?

Kind regards,
Oliver

**Comment by LinfengLiu98** [comment]
Hi,

These points that you disagree with, did you put them in the README or anywhere in the report? If you didn't put them in, you cannot add anything into the README by now. I checked it again and I really cannot see where you put the results. I can see in the code you have train, val, and test split but in the report, there is no plot of training, val acc/loss and the proof of tesing results. And no hyperparameter tuning results were presented. I will remove the previous comments "No train, val and test split.", but still there is no plot for train and val acc/ loss and how you obtained the test acc.

Kind regards,
Linfeng

**Comment by o-20** [comment]
Hi,

Why am I not able to add anything to the README now? It says that I have a
week to respond.

The test accuracy was obtained the same way as the other accuracy, using
the test function. It should be visible from the code.

There was hyperparameter tuning, it mightn't have been included in the
readme I would have to check again. Actually there was extensive
hyperparameter tuning, for over a week of testing.

Kind regards,
Oliver

**Comment by LinfengLiu98** [comment]
Hi Oliver,

You can change if you want and Shakes will check the commit logs to give the final score. I don't see where it's been mentioned in the task sheet saying you have a week to change but sure, feel free to add what you think it's necessary :).

Kind regards,
Linfeng

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage, no patient level split -1
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, no dependencies -1
Model/technical explanation, could be better -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required, done -2 +2
Request Description minimal -1

**Comment by o-20** [comment]
Hi,

I am not sure what is meant by no patient level split. Could that please be
explained to me because I am sure I have it.

Kind regards,
Oliver
